exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/transactions/ThreadLocalTransactionManager.kt

```diff
@@ -60,15 +66,23 @@ class ThreadLocalTransactionManager(
 
     private class ThreadLocalTransaction(
         override val db: Database,
+        private val setupTxConnection: ((ExposedConnection<*>, TransactionInterface) -> Unit)?,
         override val transactionIsolation: Int,
+        override val readOnly: Boolean,
         val threadLocal: ThreadLocal<Transaction>,
         override val outerTransaction: Transaction?
     ) : TransactionInterface {
 
         private val connectionLazy = lazy(LazyThreadSafetyMode.NONE) {
             outerTransaction?.connection ?: db.connector().apply {
-                autoCommit = false
-                transactionIsolation = this@ThreadLocalTransaction.transactionIsolation
+                setupTxConnection?.invoke(this, this@ThreadLocalTransaction) ?: run {
+                    // The order of `setReadOnly` and `setAutoCommit` is important.
+                    // Some drivers start a transaction right after `setAutoCommit(false)`,
+                    // which makes `setReadOnly` throw an exception if it is called after `setAutoCommit`
+                    isReadOnly = outerTransaction?.readOnly ?: this@ThreadLocalTransaction.readOnly
```

Comment:
I think `outerTransaction?.readOnly` is always null here as `outerTransaction?.connection` can be null only if there is no outerTransaction